### ISSUE TITLE
Wrong output registered when granting privileges on specific columns

### ISSUE THREAD

**Issue by fernandocarletti** [pulumi/pulumi-postgresql#384]
### What happened?

When granting access to specific columns the provider does not register the outputs correctly causing the Grant resource to be recreated indefinetely.indefinitely

### Example

```typescript
let database = new postgresql.Database("pulumi-db");

let role = new postgresql.Role('foo');

new postgresql.Grant('foo', {
  database: database.name,
  role: role.name,
  schema: 'information_schema',
  objectType: 'column',
  objects: ['sql_features'],
  columns: ['feature_id', 'feature_name'],
  privileges: ['SELECT']
});
```

The code above was tested with `docker run -ti --rm -p 5432:5432 -e POSTGRES_PASSWORD=password postgres:16`. Every update will cause the Grant resource to be recreated.

### Output of `pulumi about`

CLI          
Version      3.105.0
Go Version   go1.21.6
Go Compiler  gc

Host     
OS       arch
Version  "23.1.3"
Arch     x86_64

_The backend part of the about was omitted since it seems not relevant to the issue._

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by fernandocarletti** [comment]
I tried the same using the terraform provider, and it works fine:

```hcl
terraform {
  required_providers {
    postgresql = {
      source = "cyrilgdn/postgresql"
    }
  }
}


provider "postgresql" {
  host            = "localhost"
  port            = 5432
  database        = "postgres"
  username        = "postgres"
  password        = "password"
  sslmode         = "disable"
  connect_timeout = 16
}

resource "postgresql_database" "foo" {
  name = "tf_foo"
}

resource "postgresql_role" "foo" {
  name = "tf_foo_role"
}

resource "postgresql_grant" "foo" {
  database    = postgresql_database.foo.name
  role        = postgresql_role.foo.name
  schema      = "information_schema"
  object_type = "column"
  objects     = ["sql_features"]
  columns     = ["feature_id", "feature_name"]
  privileges  = ["SELECT"]
}
```


Running a verbose update (`-v5 --logflow --logtosdterr`) I could see that at some point the output is return but in a second log the output is empty for that attribute. I'm not very familiar with the bridge to figure out where it is getting lost exactly.

**Comment by iwahbe** [comment]
Hi @fernandocarletti. Thanks for reporting this bug and the excellent repro! We'll take a look as soon as we can.

**Comment by lukehoban** [comment]
I just tried reproing this, and it no longer repros.  I confirmed that it repros as recently as the [3.11.1 release](https://github.com/pulumi/pulumi-postgresql/releases/tag/v3.11.1), and was fixed in the [3.11.3 release](https://github.com/pulumi/pulumi-postgresql/releases/tag/v3.11.3).  

It appears the fix was in improvements in pulumi-terraform-bridge as part of this release: https://github.com/pulumi/pulumi-terraform-bridge/compare/v3.86.0...v3.87.0.

**Comment by fernandocarletti** [comment]
Awesome! Thanks for the heads up, @lukehoban!
